realm-jni/build.gradle

```diff
@@ -87,15 +87,34 @@ if (project.forceDownloadCore instanceof String) {
     ext.forceDownloadCore = Boolean.parseBoolean(project.forceDownloadCore)
 }
 
+def getNdkDir() {
+    def ndkDir = null
+    if (project.ext.properties.containsKey('ndk.dir')) {
+        ndkDir = project.ext['ndk.dir']
+    } else if (System.getenv('ANDROID_NDK')) {
+        ndkDir = System.getenv('ANDROID_NDK')
+    } else if (System.getenv('NDK_HOME')) {
+        ndkDir = System.getenv('NDK_HOME')
+    }
+    if (ndkDir) {
+        if (ndkDir[-1] == "/") {
+            ndkDir = ndkDir[0..-2]
+        }
+        def ndkVersion = new File("${ndkDir}/RELEASE.TXT").getText('UTF-8').split()[0]
+        if (ndkVersion != 'r10d') {
+            throw new GradleException("Your NDK version: ${ndkVersion}. Realm JNI should be compiled with the version r10d of NDK.")
+        }
+    }
+    return ndkDir
+}
+
 task checkProperties(group: 'check', description: 'Check the user provided gradle.properties') << {
-    if (!project.ext['ndk.dir']) {
-        throw new GradleException('The ndkDir property in the gradle.properties file is not set.')
-    } else if (project.ext['ndk.dir'].endsWith('/')) {
-        throw new GradleException('The path provided in the ndkProperty in the gradle.properties ends with /')
-    } else if (!file(project.ext['ndk.dir']).directory) {
-        throw new GradleException('The path provided in the ndkDir property in the gradle.properties is not a folder.')
-    } else if (!file(new File(project.ext['ndk.dir'], 'RELEASE.TXT')).file) {
-        throw new GradleException('The path provided in the ndkDir property in the gradle.properties does not seem to be an Android NDK.')
+    if (!getNdkDir()) {
+        throw new GradleException('The ndk.dir property in the local.properties file and ANDROID_NDK are not set. One of them should be set.')
+    } else if (!file(getNdkDir()).directory) {
+        throw new GradleException('The path provided in the ndk.dir property in the local.properties or ANDROID_NDK is not a folder.')
+    } else if (!file(new File(getNdkDir(), 'RELEASE.TXT')).file) {
```

Comment:
This check seems really strange? It is just checking if a file `RELEASE.TXT` exists, which doesn't mean if it is a ndkdir or not. Right now just calling `getNdkDir()` will make that check`

I would split `getNdkDir` into two methods `getNdkDir` and `verifyNdkDir` that would make the code and these if-statements more readable IMO